test/OSX.hs

```diff
@@ -4,6 +4,18 @@
 module OSX(sdot) where
 
 import Foreign.Ptr
+import Foreign.Marshal.Alloc
+import Foreign.Storable
 
-foreign import ccall "cblas_sdot"
-   sdot :: Int -> Ptr Float -> Int -> Ptr Float -> Int ->  Float
+foreign import ccall "sdot_"
```

Comment:
The C version of blas handled some of the marshaling (allocating pointers for the integer arguments).   Calling the FORTRAN code directly from Haskell requires marshaling the integers prior to the foreign call.